src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Members.cs

```diff
@@ -160,13 +161,21 @@ public override void VisitProperty(IPropertySymbol symbol)
                 AddPunctuation(SyntaxKind.OpenBraceToken);
 
                 AddAccessor(symbol, symbol.GetMethod, SyntaxKind.GetKeyword);
-                AddAccessor(symbol, symbol.SetMethod, SyntaxKind.SetKeyword);
+                // PROTOTYPE(init-only): adjust SymbolDisplayVisitor once we have a public IsInitOnly API
+                var keywordForSetAccessor = IsInitOnly(symbol)
+                    ? SyntaxKind.InitKeyword : SyntaxKind.SetKeyword;
+                AddAccessor(symbol, symbol.SetMethod, keywordForSetAccessor);
 
                 AddSpace();
                 AddPunctuation(SyntaxKind.CloseBraceToken);
             }
         }
 
+        private static bool IsInitOnly(IPropertySymbol symbol)
+        {
+            return (symbol.SetMethod as Symbols.PublicModel.MethodSymbol)?.UnderlyingMethodSymbol.IsInitOnly == true;
```

Comment:
>symbol.SetMethod [](start = 20, length = 16)

Consider extracting a helper that takes IMethodSymbol, so that we can use it below.